addons/hr_attendance/models/hr_employee.py

```diff
@@ -99,7 +99,7 @@ def _compute_hours_today(self):
     @api.depends('attendance_ids')
     def _compute_last_attendance_id(self):
         for employee in self:
-            employee.last_attendance_id = self.env['hr.attendance'].search([
+            employee.last_attendance_id = self.env['hr.attendance'].sudo().search([
```

Comment:
Hi. naive question : why don't you set compute_sudo in _compute_last_attendance_id ?

Reply:
Hello : ) 

I could. But I believe it's a good security practice to pinpoint exactly where you need higher privileges.
In this case, this is indeed the only line of the method, so it would basically be the same.

But who knows in the future what will happen here, what will be added, and if I add the `compute_sudo=True` on the field definition directly, the developer who will add something there might miss that fact, that's it's sudo-ed.

However, if there are modules overriding the method and require sudo, such as multiple cases pin-pointed in this pull request, the case of sale orders changed by inventory users without access to the sale orders.

So, in conclusion, if it requires it, it can be moved on the `compute_sudo=True` of the field definition, but otherwise it's better to pin-point exactly where the sudo is required, as a security good practice.

Reply:
Thanks for the complete answer ! It is very clear.
Maybe, you could add part of this explanation [here ](https://github.com/odoo/odoo/blob/master/odoo/fields.py#L261) to help developper to use the good way to sudo.